specification.md

```diff
@@ -2474,6 +2531,217 @@ authenticity of the downloaded file, for example can be the
 The exact signing and verification method is Agent specific. See
 [Code Signing](#code-signing) for recommendations.
 
+### Custom Messages
+
+Status: [Development]
+
+#### Motivation
+
+The OpAMP protocol is intended to cover all aspects of remote management and configuration
+of Agents. If more requirements within this core functionality are identified, it is
+expected that the protocol will be extended to support this additional behavior in future
+releases. However, there are some use cases that require communication with Agents that
+will not by supported by the OpAMP protocol because they are not within the scope of Agent
+remote management or are highly specific to one agent and not generalizable to other agent types. CustomCapabilities and CustomMessage allow custom behavior to be
+implemented between a Server and Agent without having to open another connection or define
+an entirely new protocol.
+
+While interoperability between many Agents and Agent Management Servers is a goal of
+OpAMP, additional custom functionality may only be supported by specific Agents and
+Servers. Where appropriate, implementers are encouraged to document their use of
+CustomMessage so that support can be added to additional platforms. If appropriate, custom
+capabilities that are widely supported by Server and Agent implementations may be adopted
+as standard OpAMP capabilities in the future.
+
+#### CustomCapabilities
+
+The Agent and Server should both use this message to signal that they support specific
+custom capabilities. It is supported on ServerToAgent and AgentToServer. This message
+should only be sent when the list of supported custom capabilities changes. When this
```

Comment:
>This message should only be sent when the list of supported custom capabilities changes.

👍 

When using an http transport it is possible for the server to observe capabilities change if the agent is updated to a new version and restarted. So this really can happen.

Note that if `ReportFullState` flag is used by the Server the Agent must report the full list of capabilities.

Can we mention `CustomCapabilities` in the "Agent Status Compression" section to make this clear?